### PR TITLE
switching to router loader data fetching approach

### PR DESCRIPTION
**Pull request by chumpy**
This PR is part of a series that upgrades our usage of react-router.

In previous PRs for this work we have:
* isolated the 21a sub app: https://github.com/department-of-veterans-affairs/vets-website/pull/33262
* installed react-router version 6 for our portal sub app: https://github.com/department-of-veterans-affairs/vets-website/pull/33339

This PR gets rid of an intermediary "fetching" component we were using previously and uses react router loaders to take advantage of the newer paradigm for data fetching that decouples component lifecycle from firing data fetching events.

### CODE REVIEW

**Comment by chumpy** [comment]
@nihil2501 I ended up removing some unit tests other than just the fetcher.  Updating the tests to use loader data approach instead of params from fetcher ended up pulling on a string that leads to upgrading some of our test libraries. I think it's worth creating a separate ticket to upgrade unit test infrastructure but also worthwhile to get this merged and integrated so we can continue to move forward.

**Comment by nihil2501** [comment]
> @nihil2501 I ended up removing some unit tests other than just the fetcher. Updating the tests to use loader data approach instead of params from fetcher ended up pulling on a string that leads to upgrading some of our test libraries. I think it's worth creating a separate ticket to upgrade unit test infrastructure but also worthwhile to get this merged and integrated so we can continue to move forward.

Awesome. Taking a look today.

**Review by jquispe-oddball — DISMISSED** [review]

File: src/applications/accredited-representative-portal/routes.jsx [hunk 1]
```diff
@@ -22,10 +25,14 @@ const router = createBrowserRouter(
...
               path: 'poa-requests',
               element: <POARequestsPage />,
+              loader: poaRequestsLoader,
+              errorElement: <ErrorMessage />,
             },
             {
               path: 'poa-requests/:id',
               element: <POARequestDetailsPage />,
+              loader: poaRequestLoader,
+              errorElement: <ErrorMessage />,
```
Comment (nihil2501) [inline src/applications/accredited-representative-portal/routes.jsx:35]: In the future we can have an error boundary hierarchy, where a generic error message like this is at the top. `POARequestDetailsPage` can handle `404` and rethrow others if the framework works that way (I think it does).

**Review by nihil2501 — CHANGES_REQUESTED** [review]
Requesting these changes now for convenience:
- Delete dead `actions/poaRequests.js`
- Go with a global spinner in the absence of React 18+'s `Suspense`
- Fix API `show` path
- Co-locate loaders with route elements (I think this is warranted for us)

Otherwise, some future-oriented comments.

File: src/applications/accredited-representative-portal/app-entry.jsx [hunk 2]
```diff
@@ -22,6 +23,8 @@ connectFeatureToggle(store.dispatch);
 
 startReactApp(
   <Provider store={store}>
-    <RouterProvider router={router} />
+    <Suspense fallback={<VaLoadingIndicator message="Loading..." />}>
+      <RouterProvider router={router} />
+    </Suspense>
```
[the remark this answers is not in the document]
Comment (nihil2501) [reply src/applications/accredited-representative-portal/app-entry.jsx:28]: Also I tested by adding this to `poaRequestLoader`:
```js
await new Promise(resolve => setTimeout(resolve, 3000));
```

File: src/applications/accredited-representative-portal/routes.jsx (hunk 1 above)
Comment (chumpy) [reply src/applications/accredited-representative-portal/routes.jsx:35]: yeah this is driven by which errors we let hot routes in the stack so down the road we can control use case vs global by what we handle vs what we throw

File: src/applications/accredited-representative-portal/app-entry.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (chumpy) [reply src/applications/accredited-representative-portal/app-entry.jsx:28]: oof yeah this isn't supported in our current versions, good catch.

**Review by nihil2501 — APPROVED** [review]
LGTM

**Pull request merged**